sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Azure.Monitory.Query.Models
+{
+    public class LogsQueryResultRow
+    {
+        private readonly Dictionary<string, int> _columns;
+        private readonly JsonElement _row;
+
+        internal LogsQueryResultRow(Dictionary<string, int> columns, JsonElement row)
+        {
+            _columns = columns;
+            _row = row;
+        }
+
+        public int Count => _row.GetArrayLength();
+
+        public int GetInt32(int index) => _row[index].GetInt32();
```

Comment:
Non-boxing overloads

Reply:
Hm, I think most of the time the access pattern would be 1x per value and materializing ints is non-allocating and cheap. The only thing that might be worth caching are strings but I'm worried that allocating another array per row might hurt more than help.